CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+cmake_minimum_required(VERSION 3.5.1)
+project(CMaizeSuperbuild VERSION 1.0.0)
+
+include(ExternalProject)
+
+include(GNUInstallDirs)
+
+set(
+        CMAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/stage"
```

Comment:
As a general rule of thumb you never want to call `set` with a variable that starts with `CMAKE`. Those variables are how users set options. If you call `set` with one of them you clobber the user's setting. In this case I recommend you call the staging directory something like `STAGING_DIRECTORY`.

Reply:
Would you recommend it be 

CMAKE_INSTALL_PREFIX "${STAGING_DIRECTORY}/stage"
CMAKE_INSTALL_PREFIX "STAGING_DIRECTORY/stage"

or remove the CMAKE_INSTALL_PREFIX with STAGING_DIRECTORY

Reply:
`set(STAGING_DIRECTORY ${CMAKE_BINARY_DIR}/stage/)` then in `ExternalProject_Add` pass the staging directory as the install directory like  `CMAKE_ARGS -DCMAKE_INSTALL_PREFIX="${STAGING_DIRECTORY}"`